### DISCUSSION TITLE
Pull comments API responds with 500 for >10,000 results

### DISCUSSION THREAD

**Discussion by fjahr** [community/community#110104] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

There seems to be either a bug or an undocumented API limit for the REST API: Retrieving all pull comments from a repository with more than 10,000 comments is currently not possible. With a `per_page` limit of 100, every `page` starting from 100 returns a 500. Similarly, with a `per_page` limit of 50, every `page` starting from 199 returns a 500.

Example:
```
$ curl "https://api.github.com/repositories/1181927/pulls/comments?page=198&per_page=50"
$ curl "https://api.github.com/repositories/1181927/pulls/comments?page=199&per_page=50"
```

This discussion might be related but it is about alerts and did not receive any responses even though it was opened in October: https://github.com/orgs/community/discussions/70012

How can this be fixed? If this is the desired behavior, why is there no documentation on this? Is there a workaround to get all comments for a large and long-running repository that works?

Thanks a lot!

**Comment by maflcko** [comment]
> How can this be fixed? If this is the desired behavior, why is there no documentation on this? Is there a workaround to get all comments for a large and long-running repository that works?

A workaround may be to list all issues/pulls instead and then get the comments for each of them?

**Reply by fjahr** [reply]
Yes, but I am trying to make a tool work that uses this endpoint in particular.
